---
title: "igv_16track"
output: html_document
date: "2022-11-21"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(tidyverse)
```

```{r}
source("https://github.com/PoisonAlien/trackplot/blob/master/R/trackplot.R?raw=true")
```

```{r}
Sys.setenv(PATH="/programs/R-4.2.1-r9/opt/bin:/programs/geos-3.11.0/bin:/programs/gdal-3.5.2/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/usr/lib/rstudio-server/bin/quarto/bin:/usr/lib/rstudio-server/bin/postback:/lustre2/home/kw572/.TinyTeX/bin/x86_64-linux:/programs/bwtool-1.0/bin/")
```

```{r}
bigWigs<-c("~/DASH_rep/Analysis/CellRanger_output/rep2_10M/outs/possorted_genome_bam.bw")
bigWigs_piwi<-c("~/DASH_rep/Analysis/CellRanger_output/rep2_100M/outs/coverage.bw")

track_data_16S <- track_extract(bigWigs = bigWigs, loci = "chrMT:21,500-23,500")
track_data_piwi <- track_extract(bigWigs = bigWigs_piwi, loci = "chr2:181,206,098-181,212,250")
```

```{r}
pdf("16S-rRNA.pdf", height=3, width = 6)
track_plot(summary_list = track_data_16S,gene_model = "~/Transcriptome/smesg_mt/genes/genes.gtf",draw_gene_track = T,isGTF = T,gene_fsize = 1,gene_track_height = 0.3,scale_track_height = 0.5,y_min = 0)
track_plot(summary_list = track_data_16S,gene_model = "~/Transcriptome/smesg_mt/genes/genes.gtf",draw_gene_track = T,isGTF = T,gene_fsize = 1,gene_track_height = 0.3,scale_track_height = 0.5,y_min = 0, y_max=10000)
track_plot(summary_list = track_data_piwi,gene_model = "~/Transcriptome/smesg_mt/genes/genes.gtf",draw_gene_track = T,isGTF = T,gene_fsize = 1,gene_track_height = 0.3,scale_track_height = 0.5,y_min = 0)

dev.off()
```


Use RSeQC: geneBody_coverage.py to generate coverages
```{r}
mt16S <- c(0.21458103638368248,0.21761300992282248,0.22243660418963615,0.2276736493936053,0.2329106945975744,0.0,0.002067254685777288,0.007855567805953693,0.0172271223814774,0.02053472987872106,0.03776185226019846,0.04534178610804851,0.05609151047409041,0.06518743109151047,0.0711135611907387,0.08710033076074973,0.100606394707828,0.11838478500551268,0.12885887541345092,0.13602535832414553,0.14250275633958104,0.1558710033076075,0.14332965821389196,0.12941014332965822,0.10887541345093715,0.137954796030871,0.1587651598676957,0.1529768467475193,0.16496692392502757,0.15270121278941565,0.12775633958103638,0.11163175303197354,0.32593715545755236,0.42351157662624034,0.4731256890848953,0.517502756339581,0.5825523704520397,0.6641400220507166,0.7523428886438809,0.8172546857772878,0.9277839029768468,0.9174476295479603,0.9863561190738699,0.6602811466372657,0.9538313120176406,0.9962789415656009,0.9805678059536935,0.982497243660419,0.9874586549062845,0.988561190738699,0.985529217199559,0.9955898566703418,0.9871830209481808,0.9775358324145534,0.9874586549062845,0.9902149944873209,0.954244762954796,0.9739525909592062,0.9796030871003307,0.9852535832414554,0.9501102535832414,0.9167585446527012,0.9400496141124587,0.9711962513781698,0.976433296582139,0.9811190738699008,0.887954796030871,0.9760198456449835,0.9768467475192943,0.9745038588754135,0.9655457552370452,0.9829106945975744,0.9517640573318633,0.9753307607497244,0.9758820286659317,0.9835997794928335,0.9877342888643881,0.8202866593164277,0.9907662624035282,0.984013230429989,0.9359151047409041,0.9783627342888643,0.960308710033076,0.9820837927232635,0.9684399117971334,0.9835997794928335,0.9797409040793826,0.8257993384785005,1.0,0.8656284454244763,0.984013230429989,0.9877342888643881,0.9790518191841234,0.9871830209481808,0.965683572216097,0.9516262403528115,0.8342061742006616,0.7125137816979051,0.6004685777287762,0.2234013230429989)

non16S<- c(0.0,0.005323286218906212,0.008392412149676244,0.011865305252936602,0.017485843377970663,0.021805904413695384,0.02623495781500032,0.03582133178762342,0.04314611417445714,0.045328438585279095,0.045048526373675374,0.046284598883323674,0.05231633547850126,0.059319094967029806,0.06441448805790467,0.07022328572348636,0.07653988872870314,0.08349558333209478,0.08831849550901912,0.09383499546690843,0.09838047253144677,0.10489028927564664,0.11286159455830844,0.11928718992910542,0.12636673949338367,0.13301775089299428,0.13849957146183534,0.14485085385610036,0.1514126896838726,0.1586433423004325,0.15985216671868574,0.16010730793811215,0.16940633840147834,0.17587899865741222,0.18366204439952638,0.18877973138336082,0.19308245272456143,0.1975362771180437,0.20510133812899742,0.21383063577228523,0.22269121967411282,0.23502965087763625,0.23407101348037393,0.24857690649941291,0.2623371926539146,0.2708608910621306,0.2785721009269305,0.3001946999985137,0.32697214254219736,0.34894648970269854,0.3603039896160001,0.37810442459462273,0.3926524283003631,0.4100540503049309,0.42936799290558786,0.44284588974926803,0.46322498501354975,0.48038137419556204,0.5020114045647984,0.5251202631670209,0.5418208660929705,0.5612239842654657,0.5762872246084945,0.590500819919841,0.603629445773821,0.6295176096983388,0.6546453041630129,0.6754207353021318,0.6885493611561118,0.7053540022492061,0.7147991815664184,0.7335855020336984,0.7509673072445243,0.7716956734985063,0.8083171083334572,0.8381116577243385,0.8554612606453339,0.8656867262161319,0.8798334398485997,0.8973093748296994,0.9136805235596906,0.92997735931315,0.9580949125336267,0.9602623743491422,0.9564104850655688,0.9972429885706642,1.0,0.9733637521117271,0.8896328443539477,0.8391099287090845,0.7776134635296682,0.7102561816010978,0.6508305713677056,0.580109388701455,0.5033762862337688,0.40797824116047143,0.3145296731715292,0.2522033797541727,0.19461082294190213,0.14155383479729897)

```

```{r}
Coverage_df<-data.frame("mt16S"=mt16S,"non16S"=non16S,percentage=1:100) %>% pivot_longer(mt16S:non16S,names_to = "gene",values_to = "value")
```

```{r}
pdf("coverage.plot.pdf",width = 6,height=4)
extrafont::loadfonts()
ggplot(Coverage_df,aes(x=percentage,y=value, color=gene))+
  geom_line()+
  theme_bw()+
  theme(text = element_text(size=12))+
  labs(x="Coverage",y="Gene body 5' -> 3'",title="non-16S loci")+
  geom_vline(xintercept = c(42,44,58),linetype="dotdash")
dev.off()

```

